kind/locals.tf

```
@@ -1,5 +1,15 @@
 locals {
-  helm_values = [{
+  helm_values = [var.distributed_mode ? {
+    loki-distributed = {
+      loki = {
+        schemaConfig  = local.schema_config
+        storageConfig = local.storage_config
+        structuredConfig = {
+          compactor = local.compactor
+        }
+      }
+    }
+    } : null, var.distributed_mode ? null : {
```

Maybe I did not understand something, but why would you do another ternary conditional if what you want can be done inside the same condition since what you want is either one or the other block of code? 

```suggestion
    } : {
```

---

I have not tested this change since I do not have a working MinIO, but theoretically it seems to me it should work in principle.

---

I'll close this conversation and approve your PR after you explained me that this was a workaround the problem that the true/false conditions were not the same type.